test/resource.cpp

```diff
@@ -51,6 +51,67 @@ TEST(CampResource, GetPlatform)
   ASSERT_EQ(static_cast<const Resource>(Omp()).get_platform(), Platform::omp_target);
 #endif
 }
+TEST(CampResource, Compare)
+{
+  Resource h1{Host()};
+  Resource h2{Host()};
+  Host h; Resource h3{h};
+
+  ASSERT_TRUE(h1 == h1);
+  ASSERT_TRUE(h1 == h2);
+  ASSERT_TRUE(h3 == h);
+
+  ASSERT_FALSE(h1 != h2);
+
+#ifdef CAMP_HAVE_CUDA
+  Resource c1{Cuda()};
+  Resource c2{Cuda()};
+  Cuda c; Resource c3{c};
+
+  ASSERT_TRUE(c1 == c1);
+  ASSERT_TRUE(c2 == c2);
+  ASSERT_TRUE(c3 == c);
+  ASSERT_TRUE(c1 != c2);
+  ASSERT_TRUE(c1 != h1);
+
+  ASSERT_FALSE(c1 == c2);
+  ASSERT_FALSE(c2 == c1);
+  ASSERT_FALSE(c1 == h1);
+  ASSERT_FALSE(c1 != c1);
```

Comment:
I'm seeing a lot of repetition, templates to the rescue?

Reply:
It seemed like templates were a bit tricky to do here, but i did reduce the repetition a bit